src/components/structures/MatrixChat.tsx

```diff
@@ -2023,6 +2023,8 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
         } else if (this.state.view === Views.LOGIN) {
             const showPasswordReset = SettingsStore.getValue(UIFeature.PasswordReset);
             const Login = sdk.getComponent('structures.auth.Login');
+            const urlParams = new URLSearchParams(window.location.search);
```

Comment:
Rather than using the "main" URL params, Element also supports reading pseudo-params from the fragment, as in `https://develop.element.io/#/login?defaultUsername=bob`. This has added advantage that it will be cleaned up by default as well once you move to the next screen.

To access this, you can read from `this.props.startingFragmentQueryParams.defaultUsername` and pass that to `Login`.

Reply:
I changed reading of the parameter to the suggested pseudo-params.
It works, but cleaning up of this params seems not to work, when moving from login to the logged in state.
For me that's okay, just wanted to mention it.

Reply:
Hmm, I am confused by this... Using the deploy preview attached to this PR (https://deploy-preview-5674--matrix-react-sdk.netlify.app), if I open a tab with https://deploy-preview-5674--matrix-react-sdk.netlify.app/#/login?defaultUsername=bob as the initial URL, I observe that the `defaultUsername` goes into the username field, and is immediately removed from URL, which becomes https://deploy-preview-5674--matrix-react-sdk.netlify.app/#/login.

@djschilling, do you see something else?